library/src/main/java/com/wultra/android/sslpinning/integration/powerauth/PowerAuthCryptoProvider.kt

```diff
@@ -35,8 +36,8 @@ class PowerAuthCryptoProvider : CryptoProvider {
 
     override fun ecdsaValidateSignatures(signedData: SignedData, publicKey: ECPublicKey): Boolean {
         val ecKey = publicKey as? PA2ECPublicKey ?: throw IllegalArgumentException("Invalid ECPublicKey object.")
-
-        return CryptoUtils.ecdsaValidateSignature(signedData.data, signedData.signature, ecKey.data)
+        val ecPublicKey = EcPublicKey(ecKey.data)
```

Comment:
This is highly ineffective. PA SDK 1.7.x allows you to create `EcPublicKey` in advance. You don't need to import the key for every signature validation. You can do the import in `importECPublicKey()` function's implementation and at this point just validate whether the key's instance is correct.